ra/ra.go

```diff
@@ -760,14 +762,27 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizationsCAA(
 // rechecked because their associated authorizations are sufficiently old and
 // performs the CAA checks required for each. If any of the rechecks fail an
 // error is returned.
-func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []string) error {
+func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, authzs []*core.Authorization) error {
 	ra.stats.Inc("recheck_caa", 1)
-	ra.stats.Inc("recheck_caa_names", int64(len(names)))
-	ch := make(chan error, len(names))
-	for _, name := range names {
-		go func(name string) {
+	ra.stats.Inc("recheck_caa_authzs", int64(len(authzs)))
+	ch := make(chan error, len(authzs))
+	for _, authz := range authzs {
+		go func(authz *core.Authorization) {
+			name := authz.Identifier.Value
+			var method string
+			for _, challenge := range authz.Challenges {
+				if challenge.Status == core.StatusValid {
+					method = challenge.Type
+					break
+				}
+			}
+			if method == "" {
+				ch <- berrors.InternalServerError("Internal error getting validation method for %v", name)
```

Comment:
Can you include the authorization ID as well as (or in place of) the name? It will be more useful for diagnosing the problem from logs.

Reply:
Done - same with the next error (existing code). I've left the name in since I suspect this could be useful where the same failure is being seen when attempting to recheck the same identifier/name across multiple authorizations.